src/main/java/Storage.java

```diff
@@ -0,0 +1,84 @@
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.ToDo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+    private static ArrayList<Task> tasks = new ArrayList<>();
+    private static final String FILEPATH = "data/savedData.txt";
+    private static final String SAVEDIR = "data";
+
+    public static void saveData(ArrayList<Task> tasks) throws IOException{
+        StringBuilder line = new StringBuilder();
+        for(Task task: tasks){
+            line.append(serializeTask(task));
+        }
+
+        FileWriter fw = new FileWriter(FILEPATH);
+        fw.write(line.toString());
+        fw.close();
+    }
+
+    public static ArrayList<Task> loadData() throws FileNotFoundException{
+        File directory = new File(SAVEDIR);
+        if(!directory.exists()){
+            if(directory.mkdir()){
+                System.out.println("Successfully created a new directory for save file");
+            }
+        } else {
```

Comment:
You can try to make the [happy path](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent) more prominent to avoid nesting these conditionals. Eg. if directory exists and direction.mkdir(), `return`. Then the rest of your code does not need to be nested in the `else` block